core/src/main/java/org/apache/iceberg/BaseFileScanTask.java

```diff
@@ -176,7 +176,7 @@ public boolean canMerge(ScanTask other) {
     @Override
     public SplitScanTask merge(ScanTask other) {
       SplitScanTask that = (SplitScanTask) other;
-      return new SplitScanTask(offset, len + that.length(), fileScanTask, deletesSizeBytes);
+      return new SplitScanTask(offset, len + that.length(), fileScanTask, deletesSizeBytes());
```

Comment:
Hm, do we actually need to force the computation of the size of deletes while merging? It seems we should keep the original logic and potentially add a comment why we are not calling `deleteSizeBytes()`. Right now, if we merge 4 tasks into 1, we will compute the size of deletes 4 times instead of 1 after the merge.

Reply:
that makes sense, I've added a comment.